instructor/common.py

```diff
@@ -200,34 +208,42 @@ def assignment_modify(request):
 def course_modify(request):
     assignment_id = request.session['assignment_id']
     assignment = get_object_or_404(models.Assignment, pk=assignment_id)
-    CourseFormSet = modelformset_factory(models.Course, extra=1, fields=['name', 'code'])
+    CourseFormSet = modelformset_factory(models.Course, extra=1, can_delete=True, fields=['name', 'code'])
     if request.method == 'POST':
-        if 'course_pk' in request.POST:
+        pk = request.POST['course_pk']
+        if not long(pk) == assignment.course.id:
             # change the course for this assignment
-            new_course = models.Course.objects.get(pk=request.POST['course_pk'])
+            new_course = get_object_or_404(models.Course, pk=long(pk))
             assignment.course = new_course
             assignment.save()
+
+        # change this course's name or code
+        formset = CourseFormSet(request.POST)
+        if formset.is_valid():
+
+            instances = formset.save(commit=False)
+            user = User.objects.get(username=request.user)
+            for obj in formset.deleted_objects:
+                if obj.owner == user:
+                    obj.delete()
+            for instance in instances:
+                if hasattr(instance, 'owner') and instance.owner != user:
```

Comment:
It turns out that when a new course form is submitted it does not have the owner attribute, since I have excluded it from the form. So 'hasattr' is checking if the course is new or not.